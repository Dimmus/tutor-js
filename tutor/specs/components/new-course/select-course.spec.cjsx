{React} = require '../helpers/component-testing'
{mount} = require 'enzyme'

SelectCourse = require '../../../src/components/new-course/select-course'
OFFERINGS = require '../../../api/offerings'

{OfferingsActions} = require '../../../src/flux/offerings'
{NewCourseStore} = require '../../../src/flux/new-course'

CourseInformation = require '../../../src/flux/course-information'

describe 'CreateCourse: Selecting course subject', ->

  beforeEach ->
    OfferingsActions.loaded(OFFERINGS)

  it 'it sets offering_id when clicked', ->
    wrapper = mount(<SelectCourse />)
    expect(NewCourseStore.get('offering_id')).not.to.exist
    wrapper.find('.list-group-item').at(0).simulate('click')
    expect(NewCourseStore.get('offering_id')).to.exist

    undefined

  it 'renders titles', ->
    wrapper = mount(<SelectCourse />)

    for offering, i in OFFERINGS
      offeringItemSelector = "[data-book-title='#{CourseInformation.forAppearanceCode(OFFERINGS.items[index].appearance_code).title}']"
      expect(wrapper.find(offeringItemSelector)).to.have.length(1)

    undefined
