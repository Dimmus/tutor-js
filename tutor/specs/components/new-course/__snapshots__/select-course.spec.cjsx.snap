exports[`CreateCourse: Selecting course subject matches snapshot 1`] = `
<div
  className="list-group">
  <div
    className="list-group-item choice active"
    data-appearance="college_biology"
    onClick={[Function]}
    role="button">
    College Biology
  </div>
  <div
    className="list-group-item choice"
    data-appearance="college_physics"
    onClick={[Function]}
    role="button">
    College Physics
  </div>
  <div
    className="list-group-item choice"
    data-appearance="intro_sociology"
    onClick={[Function]}
    role="button">
    Introduction to Sociology
  </div>
  <div
    className="list-group-item choice"
    data-appearance="ap_biology"
    onClick={[Function]}
    role="button">
    Biology for APÂ® Courses
  </div>
</div>
`;
